chalk-engine/src/logic.rs

```diff
@@ -248,36 +223,70 @@ impl<C: Context> Forest<C> {
             self.tables[table].pop_next_strand_if(|strand| strand.last_pursued_time < work)
         {
             strand.last_pursued_time = work;
-            let result: StrandResult<C, ()> = self.pursue_strand(context, depth, strand);
```

Comment:
Maybe I'll see the reason why, but it feels like this makes the enclosing function rather harder to read -- I'd rather not have nested loops in one function if we can avoid it.

Reply:
Well, I guess I can see why, we do often return straight from there